modules/microshift-cni.adoc

```diff
@@ -0,0 +1,304 @@
+//----
+//* Add the network bridge:
+//+
+//[source, terminal]
+//----
+//$ sudo ovs-vsctl add-br br-ex
+//----
+//* Add the interface to the network bridge:
+//+
+//[source, terminal]
+//----
+//$ sudo ovs-vsctl add-port br-ex <physical-interface-name>
+//----
+//The `<physical-interface-name>` is the network interface name where the node IP address is assigned.
+//* Get the bridge up and running:
+//+
+//[source, terminal]
+//----
+//$ sudo ip link set br-ex up
+//----
+//* After `br-ex up` is running, assign the node IP address to `br-ex` bridge:
+//[source, terminal]
+//----
+//$ sudo ...
+//----
+//[NOTE]
+//Adding a physical interface to `br-ex` bridge will disconnect the ssh connection on the node IP address.
+
+== Firewalld
+Explicitly allow the following OVN-Kubernetes traffic when the firewalld service is running:
+//Q: procedure?
+CNI pod to CNI pod::
+CNI pod to Host-Network pod
+Host-Network pod to Host-Network pod
+
+CNI pod::
+The Kubernetes pod that uses the CNI network
+
+Host-Network pod::
+The Kubernetes pod that uses host network
+
+=== Configuring firewalld
+.Procedure
+Insert the following firewall rules to allow the OVN-Kubernetes traffic by running the following commands.
+//Q: assumes configuring from kickstart
+* Insert the IP range:
+[source, terminal]
+----
+$ sudo firewall-offline-cmd --permanent --zone=trusted --add-source=10.42.0.0/16
+----
+
+* Insert the DNS server:
+[source, terminal]
+----
+$ sudo firewall-offline-cmd --permanent --zone=trusted --add-source=169.254.169.1
+----
+
+* Reload the firewall rules:
++
+[source, terminal]
+----
+$ sudo firewall-offline-cmd
+----
+
+== Known issue
+To avoid breaking traffic flows with a firewall restart, execute firewall commands before starting OVN-Kubernetes pods. OVN-Kubernetes makes use of iptable rules for some traffic flows (such as nodePort service). The iptable rules are generated and inserted by the `ovnkube-master` container, but are deleted when the firewall restarts. The absence of the iptable rules breaks traffic flows. If firewall commands have to be executed after OVN-Kubernetes pods have started, manually restart the `ovnkube-master` pod to trigger a reconciliation of the iptable rules.
+//See Troubleshooting for a detailed procedure. Need hard link to troubleshooting section
+
+== IP forward
+The host network `sysctl net.ipv4.ip_forward` is automatically enabled by the `ovnkube-master` container when started. This is required to forward incoming traffic to the CNI. For example, accessing the NodePort service from outside of a cluster fails if `ip_forward` is disabled.
+
+== Host ports
+Ingress routes are exposed on host ports 80 and 443 automatically by route-default deployment in the openshift-ingress namespace. Data profiling with pprof is exposed on host port 29500 when starting {product-title} with `--debug.pprof` disabled by default.
+//Q: should Network topology be in the assembly intro? seems like networking as a whole, not CNI per se
```

Comment:
Not sure if user would be interested in the detailed topology info, like in line 225, it explains how traffic is steered through ovs bridges. I suggest we simplify this to only contains info that is directly visible to user, such as the snapshot of ovs bridges.